app/src/main/java/com/moviefiend/torian/moviefiend/StarView.java

```diff
@@ -0,0 +1,94 @@
+package com.moviefiend.torian.moviefiend;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.Path;
+import android.graphics.Region;
+import android.support.annotation.ColorInt;
+import android.support.annotation.FloatRange;
+import android.util.AttributeSet;
+import android.view.View;
+
+public class StarView extends View {
+
+    private final Paint mLinePaint;
+    private final Paint mFillPaint;
+    private final Path mStarLinePath;
+    private final Path mStarFillPath;
+
+    private @FloatRange(from=0.0, to=1.0) float mPercent;
+
+    public StarView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+
+        mLinePaint = new Paint();
+        mLinePaint.setStrokeWidth(10);
+        mLinePaint.setStyle(Paint.Style.STROKE);
+        mLinePaint.setColor(Color.BLACK);
+
+        mFillPaint = new Paint();
+        mFillPaint.setStyle(Paint.Style.FILL);
+        mFillPaint.setColor(Color.YELLOW);
+
+        mStarLinePath = new Path();
+        mStarFillPath = new Path();
+
+        mPercent = 0f;
+    }
+
+    @Override
+    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+        super.onSizeChanged(w, h, oldw, oldh);
+
+        createStarPath(mStarLinePath);
+        createStarPath(mStarFillPath);
```

Comment:
@ifeins I added support for padding in the path method, but since this also changes the star's size I don't think that padding is a good property for this view anyway :)